src/config.py

```diff
@@ -63,10 +62,12 @@ def __post_init__(self):
     def file_content(self) -> dict | list:
         return (
             JsonFileType.load_from_file(
-                Path(ROOT_PATH) / self.destination_file_location.substituted_path
+                self.destination_file_location.root_path
+                / self.destination_file_location.substituted_path
```

Comment:
confused how this works

Reply:
I have added `root_path` as an attribute of `FileLocation`. This class has the `destination_file_location` attribute of type `FileLocation`. So from this attribute we obtain the root path. The same attribute also has fully resolved (and potentially substituted) paths.

However as I write this, that is a relative path FROM the root path. E.g root path of `/foo` and substituted path of `bar/baz.json` tell the program the file is at `/foo/bar/baz.json`.

As I write this, I'm realising now that the `FileLocation` contains the root path, we should just set the substituted path to be full paths I think. From this point onwards in the code I think we are just constantly merging the root path and any relative paths.

What do you think?

And does this clear things up?

Reply:
@james-ecd ?

Reply:
I meant I'm not sure how the forward slashes in between those strings is valid syntax

Reply:
https://stackoverflow.com/questions/48190959/how-do-i-append-a-string-to-a-path-in-python

Reply:
Wagwan